crates/anvil-polkadot/src/substrate_node/lazy_loading/backend/mod.rs

```diff
@@ -0,0 +1,482 @@
+mod block_import_operation;
+mod blockchain;
+mod forked_lazy_backend;
+
+pub use block_import_operation::BlockImportOperation;
+pub use blockchain::Blockchain;
+pub use forked_lazy_backend::ForkedLazyBackend;
+
+#[cfg(test)]
+mod tests;
+
+use polkadot_sdk::{
+    sc_client_api::{
+        HeaderBackend, TrieCacheContext, UsageInfo,
+        backend::{self, AuxStore},
+    },
+    sp_blockchain,
+    sp_core::{H256, offchain::storage::InMemOffchainStorage},
+    sp_runtime::{
+        Justification, StateVersion,
+        traits::{Block as BlockT, Header as HeaderT, NumberFor, One, Saturating, Zero},
+    },
+};
+use serde::de::DeserializeOwned;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+
+use crate::substrate_node::lazy_loading::rpc_client::RPCClient;
+
+pub struct Backend<Block: BlockT + DeserializeOwned> {
+    pub(crate) rpc_client: Option<Arc<dyn RPCClient<Block>>>,
+    pub(crate) fork_checkpoint: Block::Header,
```

Comment:
this should be optional, granted that we can use the lazy loading backend without forking